### ISSUE TITLE
no-static-element-interactions - false positive

### ISSUE THREAD

**Issue by javaherian** [evcohen/eslint-plugin-jsx-a11y#272]
`no-static-element-interactions` correctly throws with the following code:

      <div styleName="container" onClick={this.props.onClick}>

But still fails once I add  `role="button"`

      <div styleName="container" onClick={this.props.onClick} role="button">

According to https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md , it should succeed once I add the role attribute.

**Comment by jessebeach** [comment]
I reproduced it.

This passes.
```
{ code: '<input type="button" onClick={() => void 0} />' },
``
This fails
```
{ code: '<div styleName="container" onClick={this.props.onClick} role="button">' },
```

Looking into why.

**Comment by jessebeach** [comment]
@javaherian, is the tag closed correctly in your file?

**Comment by jessebeach** [comment]
@javaherian Also, what version of the plugin are you using?

**Comment by jessebeach** [comment]
```
{ code: '<div styleName="container" onClick={this.props.onClick} role="button">' },
```

Turns out this is an invalid template for testing. it needs to be closed. @javaherian I need some more info from you to proceed. Are you sure that the rule being triggered after you add `role="button"` isn't `interactive-supports-focus`?

**Comment by javaherian** [comment]
<div styleName="container" onClick={this.props.onClick} role="button">
        </div>

fails with this error

    99:9  error  Visible, non-interactive elements should not have mouse or keyboard event listeners  jsx-a11y/no-static-element-interactions

**Comment by javaherian** [comment]
@jessebeach I'm not sure how to find the version of the plugin

**Comment by jessebeach** [comment]
@javaherian no worries. If you look in the node_modules folder for your project, you'll find this plugin. Look in the package.json file for the version field. That path is probably:

```
./node_modules/eslint-plugin-jsx-a11y/package.json
```

**Comment by javaherian** [comment]
Cool. It's ver. 2.2.3

Dara

**Comment by javaherian** [comment]
Can you reproduce this issue?

Dara

**Comment by johnpbloch** [comment]
I'm also experiencing this issue on v4.0.0. Here's the relevant code:

```js
                        <div
                            className="select-tags"
                            onClick={this.toggleTagSelector}
                            role="button">
                            <select>
                                <option>Select Tags</option>
                            </select>
                            <div
                                className="over-select"
                                onClick={e => e.preventDefault()}
                                role="presentation">
                            </div>
                        </div>
```

And the error when I run `./node_modules/.bin/eslint --fix <path/to/script>`:

```sh
  222:25  error    Visible, non-interactive elements should not have mouse or keyboard event listeners  jsx-a11y/no-static-element-interactions
  229:29  error    Visible, non-interactive elements should not have mouse or keyboard event listeners  jsx-a11y/no-static-element-interactions
```

**Comment by johnpbloch** [comment]
Upgrading to v5.1.1 resolved the issue, FWIW. There are now a lot more errors, but they seem to be true positives, rather than this issue, which was a false positive. :)

**Comment by johnpbloch** [comment]
😥 looks like `^5.0.0` isn't supported by the version of `eslint-config-airbnb` I'm using, so I can't upgrade this package past `^4.0.0` yet...

**Comment by jessebeach** [comment]
@johnpbloch bummer :( I'd recommend putting a suppression comment about the false positive and indicate in the comment that it should be removed once this dependency has moved past version 5.

https://stackoverflow.com/questions/27732209/turning-off-eslint-rule-for-a-specific-line

**Comment by jessebeach** [comment]
> 😥 looks like ^5.0.0 isn't supported by the version of eslint-config-airbnb I'm using, so I can't upgrade this package past ^4.0.0 yet...

That is a bummer :( This is personal preference of course...I don't like including configuration as a dependency. Config is often unique to a project. You could untethered your system from this dependency; just copy-paste the config into your system and then treat it as a local file from then on.

**Comment by ljharb** [comment]
Note that this hasn't been the case for awhile now; `eslint-config-airbnb` has supported v5 for awhile, and v16 of the config supports v6 of this plugin.
